File: app/src/main/java/de/tum/in/tumcampusapp/component/ui/overview/SupportCard.java
package de.tum.in.tumcampusapp.component.ui.overview;

import android.content.Context;
import android.content.Intent;
import android.content.SharedPreferences;
import android.content.SharedPreferences.Editor;
import android.net.Uri;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import de.tum.in.tumcampusapp.R;
import de.tum.in.tumcampusapp.component.ui.overview.card.Card;
import de.tum.in.tumcampusapp.component.ui.overview.card.CardViewHolder;
import de.tum.in.tumcampusapp.utils.Utils;

/**
 * Card that describes how to dismiss a card
 */
public class SupportCard extends Card {

    public SupportCard(Context context) {
        super(CardManager.CARD_SUPPORT, context, "", true);
    }

    public static CardViewHolder inflateViewHolder(ViewGroup parent) {
        final View view = LayoutInflater.from(parent.getContext())
                                        .inflate(R.layout.card_support, parent, false);

        view.findViewById(R.id.facebook_button)
            .setOnClickListener(v -> {
                Intent browserIntent = new Intent(Intent.ACTION_VIEW);
                browserIntent.setData(Uri.parse(view.getContext()
                                                    .getString(R.string.facebook_link)));
                v.getContext()
                 .startActivity(browserIntent);
            });

        view.findViewById(R.id.github_button)
            .setOnClickListener(v -> {
                Intent browserIntent = new Intent(Intent.ACTION_VIEW);
                browserIntent.setData(Uri.parse(view.getContext()
                                                    .getString(R.string.github_link)));
                v.getContext()
                 .startActivity(browserIntent);
            });

        view.findViewById(R.id.email_button)
            .setOnClickListener(v -> {
                Intent intent = new Intent(Intent.ACTION_SENDTO);
                intent.setType("text/plain");
                intent.setData(Uri.parse(view.getContext().getString(R.string.support_email_link)));
                intent.putExtra(Intent.EXTRA_SUBJECT, view.getContext().getString(R.string.feedback));
                v.getContext()
                 .startActivity(Intent.createChooser(intent, "Send Email"));
            });

        return new CardViewHolder(view);
    }

    @Override
    public void discard(Editor editor) {
        Utils.setSetting(getContext(), CardManager.SHOW_SUPPORT, false);
    }

    @Override
    protected boolean shouldShow(SharedPreferences p) {
        return Utils.getSettingBool(getContext(), CardManager.SHOW_SUPPORT, true);
    }

    @Override
    public Intent getIntent() {
        return null;
    }

    @Override
    public int getId() {
        return 0;
    }
}

File: app/src/main/java/de/tum/in/tumcampusapp/component/ui/overview/SupportCard.kt
package de.tum.`in`.tumcampusapp.component.ui.overview

import android.content.Context
import android.content.Intent
import android.content.SharedPreferences
import android.content.SharedPreferences.Editor
import android.net.Uri
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup

import de.tum.`in`.tumcampusapp.R
import de.tum.`in`.tumcampusapp.component.ui.overview.card.Card
import de.tum.`in`.tumcampusapp.component.ui.overview.card.CardViewHolder
import de.tum.`in`.tumcampusapp.utils.Utils

/**
 * Card that describes how to dismiss a card
 */
class SupportCard(context: Context) : Card(CardManager.CARD_SUPPORT, context, "") {

    public override fun discard(editor: Editor) {
        Utils.setSetting(context, CardManager.SHOW_SUPPORT, false)
    }

    override fun shouldShow(prefs: SharedPreferences) =
            Utils.getSettingBool(context, CardManager.SHOW_SUPPORT, true)

    override fun getId() = 0

    companion object {
        @JvmStatic
        fun inflateViewHolder(parent: ViewGroup, interactionListener: CardInteractionListener): CardViewHolder {
            val view = LayoutInflater.from(parent.context).inflate(R.layout.card_support, parent, false)
            view.findViewById<View>(R.id.facebook_button).setOnClickListener { v ->
                val browserIntent = Intent(Intent.ACTION_VIEW)
                browserIntent.data = Uri.parse(view.context
                        .getString(R.string.facebook_link))
                v.context.startActivity(browserIntent)
            }
            view.findViewById<View>(R.id.github_button).setOnClickListener { v ->
                val browserIntent = Intent(Intent.ACTION_VIEW)
                browserIntent.data = Uri.parse(view.context
                        .getString(R.string.github_link))
                v.context.startActivity(browserIntent)
            }
            view.findViewById<View>(R.id.email_button).setOnClickListener { v ->
                val intent = Intent(Intent.ACTION_SENDTO)
                val uri = Uri.parse(view.context.getString(R.string.support_email_link))
                intent.setDataAndType(uri, "text/plain")
                intent.putExtra(Intent.EXTRA_SUBJECT, view.context.getString(R.string.feedback))
                v.context.startActivity(Intent.createChooser(intent, "Send Email"))
            }
            return CardViewHolder(view, interactionListener)
        }
    }
}
